lib/c-api/src/wasm_c_api/engine.rs

```diff
@@ -99,6 +99,7 @@ pub struct wasm_config_t {
     engine: wasmer_engine_t,
     #[cfg(feature = "compiler")]
     compiler: wasmer_compiler_t,
+    nan_canonicalization: bool,
```

Comment:
If this is for the spec-test compliant part of nan canonicalization and not the "canonicalize everywhere" thing we also offer, I'd make it more explicit that this will make the Wasm VM behave in a non-spec compliant way. Probably in the name of the function or something, include the word "unsafe", like what React does.